pkg/provisioner/ironic/ironic.go

```diff
@@ -739,8 +739,7 @@ func (p *ironicProvisioner) setDirectDeployUpdateOptsForNode(ironicNode *nodes.N
 		"image_disk_format":   imageData.DiskFormat,
 	}
 	updater.
-		SetInstanceInfoOpts(optValues, ironicNode).
-		SetTopLevelOpt("deploy_interface", "direct", ironicNode.DeployInterface)
```

Comment:
Doing remove is by far the simplest way, at the very least it will save you an unnecessary API request.

Reply:
Well, it does not sound simpler, as per code, according to this https://github.com/metal3-io/baremetal-operator/pull/953#discussion_r697463092 there is a need to update a flow and unit tests.

Also if we add newer `deploy_interface` value (other than "custom-deploy" or "ramdisk" ) this check that you mentioned also would have to be modified.

Either way, saving unnecessary API request does not sound very beneficial vs more complex logic.

I am not apposed to the change, but it feels way complex.

Reply:
If we do:

```
SetTopLevelOpt("deploy_interface", "", ironicNode.DeployInterface)
```

that will set it to the default AIUI.

If we do:

```
SetTopLevelOpt("deploy_interface", nil, ironicNode.DeployInterface)
```

that will remove it, but with the caveat that this will result in an unconditional API request to ironic (since ironicNode.DeployInterface is a `string`, it will always mismatch with `nil`). I guess we could make it conditional by doing:

```
if ironicNode.DeployInterface != "" {
    SetTopLevelOpt("deploy_interface", nil, ironicNode.DeployInterface)
}
```

but I'm not sure that achieves anything over the first example.

None of these feels particularly complex to me.

Reply:
@zaneb Thanks for example, setting `nil` to remove value somehow slipped my mind, indeed this is pretty simple.
@dtantsur thanks.